src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(num) {
+    if (num === undefined) {
+      const result = sum;
+
+      sum = 0;
```

Comment:
Resetting the 'sum' to 0 after obtaining the result is a crucial step to ensure that subsequent calls to 'adder' start with a clean state.